src/orange/g4org/SolidConverter.cc

```diff
@@ -569,12 +580,74 @@ auto SolidConverter::torus(arg_type solid_base) -> result_type
 }
 
 //---------------------------------------------------------------------------//
-//! Convert a generic trapezoid
+/** Convert a trapezoid.
+ *
+ * Here is the full description of the G4Trap parameters, as named here:
+ *
+ * hz  - Half Z length - distance from the origin to the bases
+ * hy1 - Half Y length of the base at -pDz
+ * hy2 - Half Y length of the base at +pDz
+ * hx1 - Half X length at smaller Y of the base at -pDz
+ * hx2 - Half X length at bigger Y of the base at -pDz
+ * hx3 - Half X length at smaller Y of the base at +pDz
+ * hx4 - Half X length at bigger y of the base at +pDz
+ * Theta - Polar angle of the line joining the centres of the bases at -/+hz
+ * Phi   - Azimuthal angle of the line joining the centre of the base at -hz
+ *         to the centre of the base at +hz
+ * Alph1 - Angle between the Y-axis and the centre line of the base at -hz
+ * Alph2 - Angle between the Y-axis and the centre line of the base at +hz
+ *
+ * Note that the numbers of x,y,z parameters in the G4Trap are related to the
+ * fact that the two z-faces are parallel (separated by hz) and the 4 x-wedges
+ * (2 in each z-face) are also parallel (separated by hy1,2).
+ *
+ * Reference:
+ * https://geant4-userdoc.web.cern.ch/UsersGuides/ForApplicationDeveloper/html/Detector/Geometry/geomSolids.html#constructed-solid-geometry-csg-solids
+ */
 auto SolidConverter::trap(arg_type solid_base) -> result_type
 {
     auto const& solid = dynamic_cast<G4Trap const&>(solid_base);
-    CELER_DISCARD(solid);
-    CELER_NOT_IMPLEMENTED("trap");
+
+    real_type sin_phi{}, cos_phi{}, tan_theta{};
+#if G4VERSION_NUMBER < 1100
+    // Geant4 10.7 and earlier - axis = (sinth.cosphi, sinth.sinphi, costh)
+    // Note: both sinth,costh >= 0 since theta is in [0, pi/2] for a G4Trap
+    auto axis = solid.GetSymAxis();
+    auto sin_theta = std::sqrt(real_type(1.0) - axis.z() * axis.z());
```

Comment:
Since all the Geant4 data are doubles, let's leave these as doubles too
```suggestion
    double sin_phi{}, cos_phi{}, tan_theta{};
#if G4VERSION_NUMBER < 1100
    // Geant4 10.7 and earlier - axis = (sinth.cosphi, sinth.sinphi, costh)
    // Note: both sinth,costh >= 0 since theta is in [0, pi/2] for a G4Trap
    auto const& axis = solid.GetSymAxis();
    auto sin_theta = std::sqrt(1.0 - axis.z() * axis.z());
```